common/maven-url-handler/src/main/java/io/helidon/build/url/mvn/Handler.java

```diff
@@ -0,0 +1,498 @@
+            checkStringHealth(version, "version");
+
+            classifier = segments[3];
+            checkStringHealth(classifier, "classifier");
+
+            segments = segments[4].split(FILE_SEPARATOR);
+
+            type = checkTypeHealth(segments[0]) ? segments[0] : DEFAULT_TYPE;
+            checkStringHealth(type, "type");
+
+            buildPathFromArchive(segments);
+        }
+
+        private void parseWithClassifierOrType(String[] segments) throws MalformedURLException {
+            if (segments.length != 4) {
+                throw new MalformedURLException("Invalid. Syntax " + SYNTAX);
+            }
+
+            version = segments[2];
+            checkStringHealth(version, "version");
+
+            segments = segments[3].split(FILE_SEPARATOR);
+
+            if (checkTypeHealth(segments[0])) {
+                type = segments[0];
+            } else {
+                classifier = segments[0];
+            }
+
+            buildPathFromArchive(segments);
+        }
+
+        private void parseNoClassifierOrType(String[] segments) throws MalformedURLException {
+            if (segments.length != 3) {
+                throw new MalformedURLException("Invalid. Syntax " + SYNTAX);
+            }
+            segments = segments[2].split(FILE_SEPARATOR);
+
+            version = segments[0];
+            checkStringHealth(version, "version");
+
+            buildPathFromArchive(segments);
+        }
+
+        private void buildPathFromArchive(String[] segments) {
+            StringBuilder builder = new StringBuilder();
+            for (int i = 1; i < segments.length - 1; i++) {
+                builder.append(segments[i]).append("/");
+            }
+            builder.append(segments[segments.length - 1]);
+            pathFromArchive = builder.toString();
+        }
+
+        private void checkStringHealth(String patient, String patientName) throws MalformedURLException {
+            if (patient.trim().length() == 0) {
+                throw new MalformedURLException(String.format("Invalid %s. Syntax  %s. ", patientName, SYNTAX));
+            }
+        }
+
+        private boolean checkTypeHealth(String type) {
+            return SUPPORTED_TYPE.contains(type);
+        }
+
+        /**
+         * Returns the group id of the artifact.
+         *
+         * @return group Id
+         */
+        public String getGroupId() {
+            return groupId;
+        }
+
+        /**
+         * Returns the artifact id.
+         *
+         * @return artifact id
+         */
+        public String getArtifactId() {
+            return artifactId;
+        }
+
+        /**
+         * Returns the artifact version.
+         *
+         * @return version
+         */
+        public String getVersion() {
+            return version;
+        }
+
+        /**
+         * Returns the artifact classifier.
+         *
+         * @return classifier
+         */
+        public String getClassifier() {
```

Comment:
If this can be `null`, then it would more correct to return an Optional (as per the Helidon guidelines)